### ISSUE TITLE
Confusion about trie files

### ISSUE THREAD

**Issue by aaronzira** [parlance/pytorch-ctc#13]
I'm unclear as to what to expect from the generated trie files. I've run generate_lm_trie.py from [deepspeech.pytorch](https://github.com/SeanNaren/deepspeech.pytorch), as well as directly within a python shell, by importing pytorch_ctc. With the former, the process takes around one second (tested with: 17GB/50k-vocab/5-gram, 2.5GB/50k-vocab/3-gram, and 6.2GB/100K-vocab/5-gram KenLM binaries), producing <3kB trie files with exactly 869 lines of mostly -1s (and 0s on every third line up to line 74), with no error messages. With the latter, the process takes some 20-30 minutes, producing ~10MB files, with 3.45M lines (still of mostly -1s) for the two aforementioned 50k-vocab binaries. 

What are the expected formats and sizes of the trie files, and is there some reference against which I might compare mine?

Secondly, using the parameters quoted in the table [here](https://github.com/SeanNaren/deepspeech.pytorch/issues/76) (beam_width 100, lm_alpha 4.0, lm_beta1 0.0, lm_beta2 5.0) increases WER/CER on a test set I'm using from 26.86/10.35 with greedy or argmax decoders to 100.00/29.63. I have not gridsearched, but I haven't found any configurations that improve WER or CER. 

Is there something obvious I'm missing?

Thanks!

**Comment by aaronzira** [comment]
Nevermind, I think I found the obvious thing I was missing -- my LM is trained on lowercased words, and my deepspeech output is uppercased =X My first question remains, though, as to the expected sizes and contents of the trie files.

**Comment by Diamondfan** [comment]
I have the same problem. There seems few usage about generate_lm_trie. I get the same CER and WER with beam search decoder.

**Comment by SeanNaren** [comment]
This should be fixed now, reducing the OOV weight makes the LM stronger. I'd also suggest increasing the alpha parameter from the default incrementally to see better LM results!
